### ISSUE TITLE
error while installing apache-airflow. 

### ISSUE THREAD

**Issue by stbones** [apache/airflow#13290]
Hi 

I am getting below error while installing apache airflow. any help?

I followed below steps to install:

**1**. created virtual env. using below command and activated it
 - ```virtualenv --python=/usr/bin/python3.6 airflow_env```
 - ```source airflow_env/bin/activate```
 
 **2. ```export AIRFLOW_HOME=/airflow_env/airflow```**
 
 **3. ran below command to install airflow** 
 ```pip install “apache-airflow[celery, crypto, mysql, rabbitmq, redis]”==1.10.10 --constraint requirements-python3.6.txt```
 Note: requirements-python3.6.txt was downloaded from below url:
 https://raw.githubusercontent.com/apache/airflow/v1-10-stable/requirements/requirements-python3.6.txt

**error message:** (i have attached complete error message as well 
[installationError.txt](https://github.com/apache/airflow/files/5736439/installationError.txt)
)
  ```
_In file included from /usr/include/python3.6m/pyconfig.h:3:0,
                   from /usr/include/python3.6m/Python.h:8,
                   from pandas/src/parser/io.h:1,
                   from pandas/src/parser/io.c:1:
  /usr/include/x86_64-linux-gnu/python3.6m/pyconfig.h:657:0: note: this is the location of the previous definition
   #define HAVE_MMAP 1

  In file included from pandas/src/parser/io.h:2:0,
                   from pandas/src/parser/io.c:1:
  pandas/src/parser/tokenizer.h:270:14: warning: inline function ‘to_longlong’ declared but never defined
   int P_INLINE to_longlong(char *item, long long *p_value);
                ^~~~~~~~~~~
  x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -SOURCE=2 build/temp.linux-x86_64-3.6/pandas/parser.o build/temp.linux-x86_64-3.6/pandas/src/parser/tokenizer.o build/temp.linux-x86_64-3.6/pandas/src/parser/io.o -o build/lib.linux-x86_64u.so
  building 'pandas._sparse' extension
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/tools-and-infrastructurges/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/include/python3.6m -c pandas/src/sparse.c -o build/temp.linux-x86_64-3.6/pandas/src/sparse.o
  In file included from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1822:0,
                   from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                   from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                   from pandas/src/sparse.c:242:
  /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable _7_API_VERSION" [-Wcpp]
   #warning "Using deprecated NumPy API, disable it with " \
    ^~~~~~~
  x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -SOURCE=2 build/temp.linux-x86_64-3.6/pandas/src/sparse.o -lm -o build/lib.linux-x86_64-3.6/pandas/_sparse.cpython-36m-x86_64-linux-gnu.so
  building 'pandas._testing' extension
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/tools-and-infrastructurges/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/include/python3.6m -c pandas/src/testing.c -o build/temp.linux-x86_64-3.6/pandas/src/testing.o
  x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -SOURCE=2 build/temp.linux-x86_64-3.6/pandas/src/testing.o -lm -o build/lib.linux-x86_64-3.6/pandas/_testing.cpython-36m-x86_64-linux-gnu.so
  building 'pandas.msgpack._packer' extension
  creating build/temp.linux-x86_64-3.6/pandas/msgpack
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -D__LITTLE_ENDIAN__=1 -Ipac -I/tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/include/python3.6m -c panda64-3.6/pandas/msgpack/_packer.o
  x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -SOURCE=2 build/temp.linux-x86_64-3.6/pandas/msgpack/_packer.o -o build/lib.linux-x86_64-3.6/pandas/msgpack/_packer.cpython-36m-x86_64-linux-gnu.so
  building 'pandas.msgpack._unpacker' extension
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -D__LITTLE_ENDIAN__=1 -Ipac -I/tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/include/python3.6m -c panda6_64-3.6/pandas/msgpack/_unpacker.o
  pandas/msgpack/_unpacker.cpp: In function ‘PyObject* __pyx_pf_6pandas_7msgpack_9_unpacker_2unpackb(PyObject*, PyObject*, PyObject*, PyObject*, int, PyObject*, PyObject*, PyObject*, PyObssize_t, Py_ssize_t)’:
  pandas/msgpack/_unpacker.cpp:2007:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       __pyx_t_2 = ((__pyx_v_off < __pyx_v_buf_len) != 0);
                     ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
  pandas/msgpack/_unpacker.cpp: In function ‘PyObject* __pyx_f_6pandas_7msgpack_9_unpacker_8Unpacker_read_from_file(__pyx_obj_6pandas_7msgpack_9_unpacker_Unpacker*)’:
  pandas/msgpack/_unpacker.cpp:3987:19: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (((__pyx_t_2 < __pyx_t_3) != 0)) {
           ~~~~~~~~~~^~~~~~~~~~~
  pandas/msgpack/_unpacker.cpp: In function ‘PyObject* __pyx_pf_6pandas_7msgpack_9_unpacker_8Unpacker_8read_bytes(__pyx_obj_6pandas_7msgpack_9_unpacker_Unpacker*, Py_ssize_t)’:
  pandas/msgpack/_unpacker.cpp:4614:19: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (((__pyx_t_1 < __pyx_t_2) != 0)) {
           ~~~~~~~~~~^~~~~~~~~~~
  In file included from pandas/msgpack/../src/msgpack/unpack.h:278:0,
                   from pandas/msgpack/_unpacker.cpp:253:
  pandas/msgpack/../src/msgpack/unpack_template.h: In function ‘int unpack_execute(unpack_context*, const char*, size_t, size_t*) [with bool construct = false]’:
  pandas/msgpack/../src/msgpack/unpack_template.h:346:9: warning: ‘obj’ may be used uninitialized in this function [-Wmaybe-uninitialized]
           c->map_key = obj;
           ^
  x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -SOURCE=2 build/temp.linux-x86_64-3.6/pandas/msgpack/_unpacker.o -o build/lib.linux-x86_64-3.6/pandas/msgpack/_unpacker.cpython-36m-x86_64-linux-gnu.so
  building 'pandas.json' extension
  creating build/temp.linux-x86_64-3.6/pandas/src/ujson
  creating build/temp.linux-x86_64-3.6/pandas/src/ujson/python
  creating build/temp.linux-x86_64-3.6/pandas/src/ujson/lib
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Ipandas/src/ujson/python Ipandas/src/klib -Ipandas/src -I/tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/ujson.c -o build/temp.linux-x86_64-3.6/pandas/src/ujson/python/ujson.o -D_GNU_SOURCE
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Ipandas/src/ujson/python Ipandas/src/klib -Ipandas/src -I/tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include -I/tools-and-infrastructure/airflow/airflow_env/include -I/usr/objToJSON.c -o build/temp.linux-x86_64-3.6/pandas/src/ujson/python/objToJSON.o -D_GNU_SOURCE
  In file included from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1822:0,
                   from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                   from /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                   from pandas/src/ujson/python/objToJSON.c:40:
  /tools-and-infrastructure/airflow/airflow_env/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable _7_API_VERSION" [-Wcpp]
   #warning "Using deprecated NumPy API, disable it with " \
    ^~~~~~~
  pandas/src/ujson/python/objToJSON.c: In function ‘initObjToJSON’:
  pandas/src/ujson/python/objToJSON.c:201:10: error: ‘NUMPY_IMPORT_ARRAY_RETVAL’ undeclared (first use in this function); did you mean ‘NPY_INOUT_ARRAY’?
     return NUMPY_IMPORT_ARRAY_RETVAL;
            ^~~~~~~~~~~~~~~~~~~~~~~~~
            NPY_INOUT_ARRAY
  pandas/src/ujson/python/objToJSON.c:201:10: note: each undeclared identifier is reported only once for each function it appears in
  pandas/src/ujson/python/objToJSON.c:202:1: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  _ERROR: Failed building wheel for pandas__
```

**Comment by potiuk** [comment]
This is a problem with installing of one of the dependencies - pandas not airflow. It needs to be built locally because apparently none of te binary packages available via PIP are matching your operating system.

You need to look for installing pandas on your system. We can't help too much. 

Moreover - I think you should consider installing either latest previous (1.10.14) or even the latest stable (2.0.0) version of Airflow (the latter preferred). 1.10.10 is already quite old and we are not actively developing 1.10.* line any more. All the new development and fixes are implemented in 2.0 line. Also there  were quite a few critical security fixes in 1.10.* line (most recent fixed in 1.10.14) so installing 1.1.0.10 now is a very bad idea.

**Comment by potiuk** [comment]
Closing as invalid for now.
